### PR TITLE
Regenerate crypto materials for mTLS testing using a script

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

File: extra/mtls/certs/README.md [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
-- `client` - example client certs, using different algorithms, signed by `tenant-ca`
+# Example certificates for the mTLS setup
+
+This folder contains the example certificates for the mTLS setup:
+
+* `/server` - example self-signed server cert
+* `tenant-ca` - example self-signed tenant certificate authority ( key pass: root)
+* `client` - example client certs, using different algorithms, signed by `tenant-ca`
+
+You can create your own certificates following the [Mutual TLS documentation](https://docs.mender.io/2.7/server-integration/mutual-tls-authentication).
```
Comment (lluiscampos) [inline extra/mtls/certs/README.md:9]: This link to documentation is tricky. It cannot point to hosted (we don't have this doc there) but pointing it to 2.7 will make it deprecated soon. I don't have a solution... :thinking:

File: extra/mtls/certs/client/cert.conf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[req]
```
Comment (lluiscampos) [inline extra/mtls/certs/client/cert.conf:1]: File committed by mistake? Is it also generated in the script

File: extra/mtls/certs/tenant-ca/tenant.ca.srl [hunk 3]
```diff
@@ -0,0 +1 @@
+0D48525F87BC581B1DD7ECB336E53F2A8C73ADA0
```
Comment (lluiscampos) [inline extra/mtls/certs/tenant-ca/tenant.ca.srl:1]: What are these `srl` finger prints?

**Review by lluiscampos — APPROVED** [review]
I am not the most openssl knowledgeable from the team, but given that we had no documentation on how to generate these previously... they look good to me :)

Thank you very much for the extra documentation/script effort.

File: extra/mtls/certs/client/cert.conf (hunk 2 above)
Comment (tranchitella) [reply extra/mtls/certs/client/cert.conf:1]: Actually, I wanted it to be in thee commit to have the right configuration used to generate the certs together with the certs. Is it a bad idea?

File: extra/mtls/certs/tenant-ca/tenant.ca.srl (hunk 3 above)
Comment (tranchitella) [reply extra/mtls/certs/tenant-ca/tenant.ca.srl:1]: Registry of signatures from the CA

File: extra/mtls/certs/README.md (hunk 1 above)
Comment (tranchitella) [reply extra/mtls/certs/README.md:9]: Same here, no idea on how to fix it.

File: extra/mtls/certs/client/cert.conf (hunk 2 above)
Comment (lluiscampos) [reply extra/mtls/certs/client/cert.conf:1]: I don't know. This file is overridden three times by the script, so why save only one of them? And why not to save also the server conf file?

These kind of questions makes me thing that it is best not to have it committed.

Comment (tranchitella) [reply extra/mtls/certs/client/cert.conf:1]: ok, let's remove them!

**Comment by tranchitella** [comment]
The pipeline was green, I force-pushed to remove openssl x509 req config files which are not used anywhere.
I won't run again the pipeline, merging it as it is.

**Pull request merged**